x/evm/keeper/grpc_query.go

```diff
@@ -409,10 +416,24 @@ func (k Keeper) TraceTx(c context.Context, req *types.QueryTraceTxRequest) (*typ
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "failed to load evm config: %s", err.Error())
 	}
+
+	// compute and use base fee of the height that is being traced
+	baseFee := k.feeMarketKeeper.CalculateBaseFee(ctx)
+	if baseFee != nil {
+		cfg.BaseFee = baseFee
+	}
+
 	signer := ethtypes.MakeSigner(cfg.ChainConfig, big.NewInt(ctx.BlockHeight()))
 
 	txConfig := statedb.NewEmptyTxConfig(common.BytesToHash(ctx.HeaderHash().Bytes()))
+
+	// store initial gas and reset gas meter per transaction
+	// to avoid stacking the gas used of every predecessor in the same gas meter
+	initialGas := ctx.GasMeter().GasConsumed()
+
 	for i, tx := range req.Predecessors {
+		// reset gas meter for each transaction
+		k.ResetGasMeterAndConsumeGas(ctx, initialGas)
```

Comment:
the main tx at line 451 (after predecessor) need to be reset as well

Reply:
Right, thanks for spotting it out Victor! Seems that we don't have a test to check proper gas consumption calculation